trigger/mqtt/trigger.go

```diff
@@ -0,0 +1,167 @@
+package mqtt
+
+import (
+	"context"
+	"encoding/json"
+	"time"
+
+	"github.com/eclipse/paho.mqtt.golang"
+	"github.com/project-flogo/core/data/coerce"
+	"github.com/project-flogo/core/data/metadata"
+	"github.com/project-flogo/core/support/log"
+	"github.com/project-flogo/core/trigger"
+)
+
+var triggerMd = trigger.NewMetadata(&Settings{}, &HandlerSettings{}, &Output{})
+
+func init() {
+	_ = trigger.Register(&MqttTrigger{}, &Factory{})
+}
+
+// MqttTrigger is simple MQTT trigger
+type MqttTrigger struct {
+	handlers []clientHandler
+	settings *Settings
+	logger   log.Logger
+}
+type clientHandler struct {
+	client mqtt.Client
+	topic  string
+	qos    int
+}
+type Factory struct {
+}
+
+func (*Factory) Metadata() *trigger.Metadata {
+	return triggerMd
+}
+
+// New implements trigger.Factory.New
+func (*Factory) New(config *trigger.Config) (trigger.Trigger, error) {
+	s := &Settings{}
+
+	err := metadata.MapToStruct(config.Settings, s, true)
+	if err != nil {
+		return nil, err
+	}
+
+	return &MqttTrigger{settings: s}, nil
+}
+
+// Initialize implements trigger.Initializable.Initialize
+func (t *MqttTrigger) Initialize(ctx trigger.InitContext) error {
+	t.logger = ctx.Logger()
+	for _, handler := range ctx.GetHandlers() {
+		options := initClientOption(t.settings)
```

Comment:
do we need a client for each handler? can one be shared?  We'll have to validate it